### ISSUE TITLE
train pipline for custom dataset

### ISSUE THREAD

**Issue by becauseofAI** [jiawei-ren/BalancedMetaSoftmax-Classification#5]
Thx for your jobs.

Can you describe the training pipline for custom dataset clearly? @jiawei-ren

**Comment by jiawei-ren** [comment]
You may modify the `LT_Dataset` class [here](https://github.com/jiawei-ren/BalancedMetaSoftmax-Classification/blob/main/data/dataloader.py#L67) to use a custom dataset.

**Comment by becauseofAI** [comment]
> You may modify the `LT_Dataset` class [here](https://github.com/jiawei-ren/BalancedMetaSoftmax-Classification/blob/main/data/dataloader.py#L67) to use a custom dataset.

@jiawei-ren 
I mean what is the meaning of the pipeline that you described as follow:
## Training
### End-to-end Training
- Base model (Representation Learning)
```bash
python main.py --cfg ./config/CIFAR10_LT/softmax_imba200.yaml
```
Alternatively, you may download a pretrained model [here](https://drive.google.com/file/d/1laY5ce0-sw3HrHBQZ2Gseo9acbG2CavK/view?usp=sharing) and put it in the corresponding log folder. 
- Balanced Softmax
```bash
python main.py --cfg ./config/CIFAR10_LT/balanced_softmax_imba200.yaml
```
### Decoupled Training
After obataining the base model, train the classifier with the following command:
- Balanced Softmax
```bash
python main.py --cfg ./config/CIFAR10_LT/decouple_balanced_softmax_imba200.yaml
```
- BALMS
```bash
python main.py --cfg ./config/CIFAR10_LT/balms_imba200.yaml
```  

## **Does that mean there are three kinds of training method?**
**Method 1: End-to-end Training**
step 1:
- Base model (Representation Learning)
```bash
python main.py --cfg ./config/CIFAR10_LT/softmax_imba200.yaml
```
Alternatively, you may download a pretrained model [here](https://drive.google.com/file/d/1laY5ce0-sw3HrHBQZ2Gseo9acbG2CavK/view?usp=sharing) and put it in the corresponding log folder. 
step 2:
- Balanced Softmax
```bash
python main.py --cfg ./config/CIFAR10_LT/balanced_softmax_imba200.yaml
```  

**Method 2: Decoupled Training**
step 1:
- Base model (Representation Learning)
```bash
python main.py --cfg ./config/CIFAR10_LT/softmax_imba200.yaml
```
Alternatively, you may download a pretrained model [here](https://drive.google.com/file/d/1laY5ce0-sw3HrHBQZ2Gseo9acbG2CavK/view?usp=sharing) and put it in the corresponding log folder. 

step 2:
- Balanced Softmax
After obataining the base model, train the classifier with the following command:
```bash
python main.py --cfg ./config/CIFAR10_LT/decouple_balanced_softmax_imba200.yaml
```

**Method 3: Decoupled Training**
step 1:
- Base model (Representation Learning)
```bash
python main.py --cfg ./config/CIFAR10_LT/softmax_imba200.yaml
```
Alternatively, you may download a pretrained model [here](https://drive.google.com/file/d/1laY5ce0-sw3HrHBQZ2Gseo9acbG2CavK/view?usp=sharing) and put it in the corresponding log folder. 

step 2:
- BALMS
After obataining the base model, train the classifier with the following command:
```bash
python main.py --cfg ./config/CIFAR10_LT/balms_imba200.yaml
```  

**If this is true, which method works better or is more recommended?**
**If not, what is the correct one?**

**Comment by jiawei-ren** [comment]
> Does that mean there are three kinds of training method?

It is mostly true, except that there is no step 1 in Method 1. End-to-end training doesn't require a pre-trained backbone.

> If this is true, which method works better or is more recommended?

Method 3 is the most recommended. You may refer to the results reported in the README or the paper.

**Comment by becauseofAI** [comment]
> > Does that mean there are three kinds of training method?
> 
> It is mostly true, except that there is no step 1 in Method 1. End-to-end training doesn't require a pre-trained backbone.
> 
> > If this is true, which method works better or is more recommended?
> 
> Method 3 is the most recommended. You may refer to the results reported in the README or the paper.

Got it. Thx u very much! @jiawei-ren

**Comment by cunjunyu** [comment]
I will close the issue for now. Feel free to reopen it if you have any further questions.
